connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -289,6 +301,38 @@ private void checkAndPutConnectorConfigName(String connectorName, Map<String, St
         return completeOrForwardRequest(cb, path, method, body, null, new IdentityTranslator<T>(), forward);
     }
 
+    private <T> T loadConnectorClass(final String className, final Class<T> type) {
+        try {
+            return type.cast(Class.forName(className).newInstance());
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    // go through config parameters and replace password field value with "*"
+    // this return a new map, instead of making change to the original object
+    private Map<String, String> maskCredentials(Map<String, String> config) {
+        final String connectorClassStr = "connector.class";
+        if (!config.containsKey(connectorClassStr)) {
+            return config;
+        }
+
+        Connector connectorClass = loadConnectorClass(config.get(connectorClassStr), Connector.class);
+        Map<String, ConfigDef.ConfigKey> definedConfigKeys = connectorClass.config().configKeys();
+        Map<String, String> newConfig = new HashMap<>();
+
+        for (Map.Entry<String, String> entry : config.entrySet()) {
+            final String key = entry.getKey();
+            if (definedConfigKeys.containsKey(key) && definedConfigKeys.get(key).type.equals(ConfigDef.Type.PASSWORD)) {
+                newConfig.put(key, "*********");
```

Comment:
Minor nit: why 9 asterisks, and not 8 or 10?

Reply:
BTW, the `Password` class is what's used within the AbstractConfig instances to hold onto a password literal so that when it is printed in the log the password is masked. It defines a `Password.HIDDEN` public static string that could be used here instead of the asterisks, and it would be more consistent with how they're hidden in the logs. WDYT?

Reply:
Great. That's what I'm looking for. Done!